python/nav/web/auth/utils.py

```diff
@@ -52,7 +57,8 @@ def ensure_account(request):
         # Assumes nobody has locked it..
         account = Account.objects.get(id=Account.DEFAULT_ACCOUNT)
 
-    _set_account(request, account)
+    # Do not cycle to avoid session_id being changed on every request
+    set_account(request, account, cycle_session_id=False)
```

Comment:
`ensure_account` basically has the task of setting `request.account` to the logged in account (decided by `ACCOUNT_ID_VAR` in the session) or, if there is no logged in account, set `Account.DEFAULT_ACCOUNT` as the logged in account. From what I understand the default account is the same as not being logged in, its more of a formality that some account is always used whether you are actually logged in or not. 

So this function does not cycle the session_id. Even though it does actually mess with the session somewhat, it doesnt actually log into a different real account

Reply:
There may be a use case for cycling the ID in part of this code: Line 55 checks if the account has been locked and effectively performs a logout if that has occurred.  I guess the scenario here is that an admin can lock a user account, and if that user account currently has an active session, they will be forcibly logged out on their next request.

Reply:
Yes, I did originally have a distinction where it cycled the ID if that type of logout occurred, but I decided to split the logout related stuff out of this PR and have that be separate PR, and have this one only focus on login

Reply:
Having a proper logout here before activating the default account is something I can do in a separate PR

Reply:
Another PR is fine by me :)

Reply:
#2828 related